presto-main/src/main/java/com/facebook/presto/operator/SortedPositionLinks.java

```diff
@@ -152,13 +153,14 @@ public Factory build()
             return new Factory()
             {
                 @Override
-                public PositionLinks create(Optional<JoinFilterFunction> lessThanFunction)
+                public PositionLinks create(Optional<JoinFilterFunction> lessThanFunction, List<JoinFilterFunction> filterFunctions)
```

Comment:
We use the lessThanFunction (`a<b AND b< a+10` in case of range predicates), for the `next() ` method. I can get rid of it and use filterFunctions ({`a< b`, `b< a+10`)  in a loop in `next()` method too (like I do in `create()`).